lms/djangoapps/teams/csv.py

```diff
@@ -406,21 +412,30 @@ def is_FERPA_bubble_breached(self, teamset_id, team_name):
         else:
             return True
 
-    def validate_proposed_team_size_wont_exceed_maximum(self, team_name, teamset_id):
+    def validate_team_sizes_not_exceeded(self):
         """
         Validates that the number of users we want to add to a team won't exceed maximum team size.
         """
-        if self.course.teams_configuration.teamsets_by_id[teamset_id].max_team_size is None:
-            max_team_size = self.course.teams_configuration.default_max_team_size
-        else:
-            max_team_size = self.course.teams_configuration.teamsets_by_id[teamset_id].max_team_size
+        for teamset_id in self.teamset_ids:
+            # Get max size for team-set
+            if self.course.teams_configuration.teamsets_by_id[teamset_id].max_team_size is None:
+                max_team_size = self.course.teams_configuration.default_max_team_size
+            else:
+                max_team_size = self.course.teams_configuration.teamsets_by_id[teamset_id].max_team_size
+
+            # Get teams in team-set
+            team_names = [
+                teamset_to_team[1] for teamset_to_team in self.user_count_by_team
+                if teamset_to_team[0] == teamset_id
+            ]
+
+            # Calculate proposed team size and return False if it exceeds capacity
+            for team_name in team_names:
+                key = (teamset_id, team_name)
+                if self.user_count_by_team[key] > max_team_size:
+                    self.add_error_and_check_if_max_exceeded('Team {} is full.'.format(team_name))
```

Comment:
I still think we should reconsider this error message. I think it will be confusing if a team is known to be full and then we get this error message.

Reply:
Updated